core/src/main/scala/kafka/controller/PartitionStateMachine.scala

```diff
@@ -146,18 +146,17 @@ class PartitionStateMachine(controller: KafkaController) extends Logging {
     val topicAndPartition = TopicAndPartition(topic, partition)
     val currState = partitionState.getOrElseUpdate(topicAndPartition, NonExistentPartition)
     try {
+      assertValidTransition(topicAndPartition, targetState)
       targetState match {
         case NewPartition =>
           // pre: partition did not exist before this
```

Comment:
Hmm, should we remove all the `pre` comments? It seems like they duplicate what's encoded as the `validPreviousStates`. Also, if we want to add comments, we should add them in `validPreviousStates` as they are less likely to get out of date there.